File: app/src/main/java/com/magdenbt/collectionsbenchmark/MainActivity.java
package com.magdenbt.collectionsbenchmark;

import androidx.fragment.app.FragmentActivity;
import androidx.lifecycle.ViewModelProvider;
import android.os.Bundle;
import com.magdenbt.collectionsbenchmark.UI.DialogCollectionSize;
import com.magdenbt.collectionsbenchmark.UI.SharedCollSizeVM;
import com.magdenbt.collectionsbenchmark.UI.VPAdapter;
import com.google.android.material.tabs.TabLayoutMediator;
import com.magdenbt.collectionsbenchmark.databinding.ActivityMainBinding;

public class MainActivity extends FragmentActivity {
    public static final String COLL_SIZE_REQ_KEY = "collestionSize";


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityMainBinding binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        SharedCollSizeVM sharedCollSizeVM = new ViewModelProvider(this).get(SharedCollSizeVM.class);

        getSupportFragmentManager().setFragmentResultListener(COLL_SIZE_REQ_KEY,this,(requestKey, result) -> sharedCollSizeVM.setCollectionsSize(result.getInt(COLL_SIZE_REQ_KEY)));
        if (sharedCollSizeVM.getCollectionSize() == 0) {
            DialogCollectionSize dialogCollectionSize = new DialogCollectionSize();
            dialogCollectionSize.setCancelable(false);
            dialogCollectionSize.show(getSupportFragmentManager(), "SomeTag");
        }

        binding.pager.setAdapter(new VPAdapter(this));
        new TabLayoutMediator(binding.tab, binding.pager,false, (tab, position)->{
           switch (position){
               case 0:
                   tab.setText(getText(R.string.tab_collections));

                   break;
               case 1:
                   tab.setText(getText(R.string.tab_maps));
                   break;
           }
        }).attach();
    }

}

File: app/src/main/java/com/magdenbt/collectionsbenchmark/ui/mainactivity/MainActivity.kt
package com.magdenbt.collectionsbenchmark.ui.mainactivity

import android.os.Bundle
import androidx.activity.compose.setContent
import androidx.fragment.app.FragmentActivity
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.get
import com.magdenbt.collectionsbenchmark.InitApp
import com.magdenbt.collectionsbenchmark.ui.sizerequestdialog.SizeRequestDialog
import com.magdenbt.collectionsbenchmark.ui.theme.AppTheme
import com.magdenbt.collectionsbenchmark.viewmodelflow.SharedCollSizeViewModel
import com.magdenbt.collectionsbenchmark.viewmodelflow.StatViewModel
import com.magdenbt.collectionsbenchmark.viewmodelflow.StatViewModelFactory
import javax.inject.Inject

class MainActivity : FragmentActivity() {

    @Inject
    lateinit var sizeRequestDialog: dagger.Lazy<SizeRequestDialog>

    @Inject
    lateinit var sharedCollSizeVM: SharedCollSizeViewModel

    companion object {
        const val COLL_SIZE_REQ_KEY = "collestionSize"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        (application as InitApp).appComponent.mainComponent().create(this).inject(this)
        super.onCreate(savedInstanceState)

        val viewModel: StatViewModel =
            ViewModelProvider(this, StatViewModelFactory(application)).get()

        supportFragmentManager.setFragmentResultListener(COLL_SIZE_REQ_KEY, this) { _, result ->
            sharedCollSizeVM.collectionSize.value = result.getInt(
                COLL_SIZE_REQ_KEY,
            )
        }

        if (sharedCollSizeVM.collectionSize.value == 0) {
            sizeRequestDialog.get().apply {
                isCancelable = false
                show(supportFragmentManager, "SomeTag")
            }
        }
        setContent {
            AppTheme {
                MainPager(viewModel = viewModel, collectionSize = sharedCollSizeVM.collectionSize)
            }
        }
    }
}
